shared/src/simulation/activities/reassign-treatments.ts

```diff
@@ -0,0 +1,632 @@
+                removeTreatmentsOfElement(draftState, patient)
+            );
+
+            const leaderId = (
+                simulatedRegion.behaviors.find(
+                    (behavior) => behavior.type === 'assignLeaderBehavior'
+                ) as AssignLeaderBehaviorState
+            )?.leaderId;
+
+            if (!leaderId) {
+                terminate();
+                return;
+            }
+
+            // The leader does not treat patients
+            personnel.splice(
+                personnel.findIndex((pers) => pers.id === leaderId),
+                1
+            );
+
+            if (personnel.length === 0) {
+                terminate();
+                return;
+            }
+
+            let allowTerminate = true;
+
+            switch (activityState.treatmentProgress) {
+                case 'unknown': {
+                    const finished = count(draftState, activityState, patients);
+                    if (finished) {
+                        sendSimulationEvent(
+                            simulatedRegion,
+                            TreatmentProgressChangedEvent.create('counted')
+                        );
+                    }
+                    allowTerminate = finished;
+                    break;
+                }
+                case 'counted': {
+                    const finished = triage(
+                        draftState,
+                        patients,
+                        personnel,
+                        materials
+                    );
+                    if (finished) {
+                        sendSimulationEvent(
+                            simulatedRegion,
+                            TreatmentProgressChangedEvent.create('triaged')
+                        );
+                    }
+                    break;
+                }
+                case 'triaged':
+                case 'secured': {
+                    const secured = treat(
+                        draftState,
+                        patients,
+                        personnel,
+                        materials
+                    );
+                    if (
+                        activityState.treatmentProgress === 'triaged' &&
+                        secured
+                    ) {
+                        sendSimulationEvent(
+                            simulatedRegion,
+                            TreatmentProgressChangedEvent.create('secured')
+                        );
+                    }
+                    break;
+                }
+                default:
+                // Unknown state
+            }
+
+            if (allowTerminate) {
+                terminate();
+            }
+        },
+    };
+
+interface CateringMaterial {
+    material: Mutable<Material>;
+    catersFor: CatersFor;
+}
+
+interface CateringPersonnel {
+    personnel: Mutable<Personnel>;
+    priority: number;
+    catersFor: CatersFor;
+}
+
+function createCateringMaterial(
```

Comment:
```suggestion
function createCateringMaterials(
```

Reply:
Fixed in https://github.com/hpi-sam/digital-fuesim-manv/pull/670/commits/db87774da41bbe59bd45ef7f524865cf1a5619e4